src/trading/getQuote.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  AccountAddress,
+  QuoteResults,
+  QuoterParameters,
+  SwapAdvancedSettings,
+  SwapParameters,
+  TradeParameters,
+} from './types'
+import { DEFAULT_QUOTE_VALIDITY, log } from './consts'
+
+import {
+  getQuoteAmountsAndCosts,
+  OrderBookApi,
+  OrderQuoteRequest,
+  OrderQuoteSideKindBuy,
+  OrderQuoteSideKindSell,
+  PriceQuality,
+  SigningScheme,
+} from '../order-book'
+import { buildAppData } from './appDataUtils'
+import { getOrderToSign } from './getOrderToSign'
+import { getIsEthFlowOrder, getSigner, swapParamsToLimitOrderParams } from './utils'
+import { Signer } from 'ethers'
+import { WRAPPED_NATIVE_CURRENCIES } from '../common'
+import { getOrderTypedData } from './getOrderTypedData'
+
+export type QuoteResultsWithSigner = { result: QuoteResults & { signer: Signer }; orderBookApi: OrderBookApi }
+
+export async function getQuote(
+  _tradeParameters: TradeParameters,
+  trader: QuoterParameters,
+  advancedSettings?: SwapAdvancedSettings,
+  _orderBookApi?: OrderBookApi
+): Promise<{ result: QuoteResults; orderBookApi: OrderBookApi }> {
+  const { appCode, chainId, account: from } = trader
+
+  const tradeParameters = getIsEthFlowOrder(_tradeParameters)
+    ? {
+        ..._tradeParameters,
+        sellToken: WRAPPED_NATIVE_CURRENCIES[chainId],
+      }
+    : _tradeParameters
+
+  const {
+    sellToken,
+    sellTokenDecimals,
+    buyToken,
+    buyTokenDecimals,
+    amount,
+    kind,
+    partnerFee,
+    validFor = DEFAULT_QUOTE_VALIDITY,
+    slippageBps = 0,
+    env = 'prod',
+  } = tradeParameters
+
+  log(`Swap ${amount} ${sellToken} for ${buyToken} on chain ${chainId}`)
+
+  const orderBookApi = _orderBookApi || new OrderBookApi({ chainId, env })
+
+  const receiver = tradeParameters.receiver || from
+  const isSell = kind === 'sell'
+
+  log('Building app data...')
+
+  const appDataInfo = await buildAppData(
+    {
+      slippageBps,
+      orderClass: 'market',
+      appCode,
+    },
+    advancedSettings?.appData
+  )
+
+  const { appDataKeccak256, fullAppData } = appDataInfo
+
+  const quoteRequest: OrderQuoteRequest = {
+    from,
+    sellToken,
+    buyToken,
+    receiver,
+    validFor,
+    appData: fullAppData,
+    appDataHash: appDataKeccak256,
+    priceQuality: PriceQuality.OPTIMAL, // Do not change this parameter because we rely on the fact that quote has id
+    signingScheme: SigningScheme.EIP712,
```

Comment:
Could this also be taken as a parameter in case we are doing a preSign?

Reply:
Added docs for the case:
https://github.com/cowprotocol/cow-sdk/pull/213/commits/b21a9f8a405002aaeef307e00140945d03ecda63